### Distributions, sampling, confidence & t-test

1. Write a for loop that prints numbers that are divisible by 2 up to 10 (incl.)


In [2]:
for i in range(1,11):
    if i%2==0:
        print(i)

2
4
6
8
10


2. Write a for loop that prints numbers from 1 to 10 (incl.), with an 80% chance of each number being printed (e.g. 1, 2, 4, 5, 7, 8, 9, 10)

In [3]:
import numpy as np

In [8]:
for i in range(1,11):
    choice = np.random.choice([0,1], p=[.2,.8])
    if choice==1:
        print(i)

1
2
3
4
5
7
9
10


3. Tossing a coin one time fits which distribution?
    - Tossing it multiple times would fit which distribution?

Tossing a coin one time fits which distribution? Bernoulli

Tossing it multiple times would fit which distribution? Binomial

4. Write a function that calculates the chances to get 60 heads or more in 100 tosses of a coin if the coin is fair. 

Hint: Do this by running a simulation.

In [22]:
def coin_flip(heads, tosses, iterations):
    iter_results = []
    for i in range(iterations):
        results = []
        for toss in range(tosses):
            result = np.random.choice([0,1], p=[.5,.5])
            results.append(result)
        iter_results.append(sum(results))
    prob_heads = sum([x>=heads for x in iter_results])/iterations
    return prob_heads

In [62]:
coin_flip(60, 100, 10000)

0.0289

In [68]:
import scipy.stats as stats
prob = stats.binom.cdf(60, 100, 0.5)
1- prob

0.01760010010885238

5. According to the exercise in question #4 explain:
    - If you were to create a hypothesis test, what would your null hypothesis be? (specifically for question #4)
    - What is your P-value? (specifically for question #4)

H0: the coin is fair (probability of 60 or more heads out of 100) = 0.0176
H1: the coin is not fair (probability of 60 or more heads out of 100) != 0.0176

6. Load the student-mat.csv dataset and perform a t-test comparing final grades of males vs females. Before running the test, remove individuals that scored 0. Use only the sex and G3 (final grade) columns to run your test.
    
    - Is there a significant difference between how women and men perform? 
    - What is your p-value? How would you interpret it?

In [43]:
import pandas as pd
df = pd.read_csv('student-mat.csv')
df.head()

school sex  age address famsize Pstatus  Medu  Fedu     Mjob      Fjob ...  \
0     GP   F   18       U     GT3       A     4     4  at_home   teacher ...   
1     GP   F   17       U     GT3       T     1     1  at_home     other ...   
2     GP   F   15       U     LE3       T     1     1  at_home     other ...   
3     GP   F   15       U     GT3       T     4     2   health  services ...   
4     GP   F   16       U     GT3       T     3     3    other     other ...   

  famrel freetime  goout  Dalc  Walc health absences  G1  G2  G3  
0      4        3      4     1     1      3        6   5   6   6  
1      5        3      3     1     1      3        4   5   5   6  
2      4        3      2     2     3      3       10   7   8  10  
3      3        2      2     1     1      5        2  15  14  15  
4      4        3      2     1     2      5        4   6  10  10  

[5 rows x 33 columns]

In [49]:
df = df[df['G3']>0]
male = df[df['sex']=='M']
female = df[df['sex']=='F']

In [55]:
from scipy.stats import ttest_ind
ttest_ind(male['G3'], female['G3'], equal_var=True)

Ttest_indResult(statistic=1.9404771273257526, pvalue=0.053113174845791505)

There is not a statistically significant difference in the final grade (G3) of male and female students at the 0.05 level of significance. The p-value is 0.053. The p-value is the probability of getting a difference between male and female test score equal to or more extreme than the value obtained from these samples if the null hypothesis of no difference in test scores between the two population was true. 